### PR TITLE
Integrate restructuring visitor

### PR DESCRIPTION
**Pull request by chris-asl**

### CODE REVIEW

File: inspire_query_parser/utils/visitor_utils.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+DATE_SPECIFIERS_REGEXES = _create_dict_of_compiled_date_regexes(DATE_SPECIFIERS_COLLECTION)
+"""Mapping from date specifier text to date specifier compiled regexes."""
+
+
+def register_date_conversion_handler(date_specifier):
+    """Decorator for registering handlers that convert text dates to dates.
+
+    Args:
+        date_specifier (str) the date specifier (in regex pattern format) for which the handler is registered
```
[the remark this answers is not in the document]
Comment (chris-asl) [reply inspire_query_parser/utils/visitor_utils.py]: Missed that one! Will do

File: inspire_query_parser/utils/visitor_utils.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+                                         DATE_SPECIFIERS_COLLECTION,
+                                         DATE_THIS_MONTH_REGEX_PATTERN,
+                                         DATE_TODAY_REGEX_PATTERN,
+                                         DATE_YESTERDAY_REGEX_PATTERN)
+
+# #### Date specifiers related utils ####
+ANY_PREFIX_AND_A_NUMBER = re.compile('(.+)(\d+)')
+
+
+def _create_dict_of_compiled_date_regexes(date_specifiers):
```
[the remark this answers is not in the document]
Comment (chris-asl) [reply inspire_query_parser/utils/visitor_utils.py]: Changing to `_compile_date_regexes` and the argument to `date_specifier_patterns`.

File: inspire_query_parser/utils/visitor_utils.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    number_match = ANY_PREFIX_AND_A_NUMBER.match(text)
+    if number_match:
+        try:
+            number = int(number_match.group(2))
+        except ValueError:
+            pass
+    return number
+
+
+def _convert_start_and_relative_delta_to_string_date(start_date, relative_delta):
```
[the remark this answers is not in the document]
Comment (chris-asl) [reply inspire_query_parser/utils/visitor_utils.py]: The optional delta is a good idea! 👍 
So, with that change in mind, the name can be `_convert_date_to_string`. How's this?

[the remark this answers is not in the document]
Comment (ammirate) [reply inspire_query_parser/utils/visitor_utils.py]: sounds ok :+1:

**Pull request merged**